test/models/item_pool_test.rb

```diff
@@ -0,0 +1,115 @@
+require "test_helper"
+
+class ItemPoolTest < ActiveSupport::TestCase
+  setup do
+    @starts = Time.zone.parse("2024-02-15")
+    @ends = Time.zone.parse("2024-02-22")
+  end
+
+  test "no items available with no reservable items" do
+    item_pool = create(:item_pool)
+
+    assert_equal 0, item_pool.max_available_between(@starts, @ends)
+  end
+
+  test "multiple items available with no reservations" do
+    reservable_item = create(:reservable_item)
+    2.times { create(:reservable_item, item_pool: reservable_item.item_pool) }
+
+    assert_equal 3, reservable_item.item_pool.max_available_between(@starts, @ends)
+  end
+
+  test "handles reservations aligned with the query" do
+    # reservation  |-------|
+    # query        |-------|
+
+    reservable_item = create(:reservable_item)
+    item_pool = reservable_item.item_pool
+    2.times { create(:reservable_item, item_pool: item_pool) }
+
+    reservation = create(:reservation, started_at: @starts, ended_at: @ends)
+    create(:date_hold, reservation: reservation, item_pool: item_pool, quantity: 1)
+
+    assert_equal 2, reservable_item.item_pool.max_available_between(@starts, @ends)
+  end
+
+  test "handles multiple reservations that don't overlap" do
+    # reservation 1  |-------|
+    # reservation 2            |-------|
+    # query               |-------|
```

Comment:
These are a great idea and help a lot with understanding - thanks for including them!